### PR TITLE
:white_check_mark: Enable all remaning assertions

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: Source/sync.cpp [hunk 1]
```diff
@@ -261,7 +261,7 @@ Uint32 sync_update(int pnum, const Uint8 *pbBuf)
 		app_fatal("bad sync command");
 	}
 
-	/// ASSERT: assert(gbBufferMsgs != BUFFER_PROCESS);
+	assert(gbBufferMsgs != BUFFER_PROCESS);
```
Comment (glebm) [inline Source/sync.cpp:264]: Windows CI failing with:
> C:\projects\devilutionx\Source\multi.cpp(469): error C2065: 'BUFFER_PROCESS': undeclared identifier [C:\projects\devilutionx\devilution.vcxproj]

Comment (AJenbo) [reply Source/sync.cpp:264]:
```suggestion
	assert(gbBufferMsgs != 2);
```

File: Source/multi.cpp [hunk 2]
```diff
@@ -466,7 +466,7 @@ void multi_process_network_packets()
 		plr[dwID]._pownerx = pkt->px;
 		plr[dwID]._pownery = pkt->py;
 		if (dwID != myplr) {
-			// ASSERT: gbBufferMsgs != BUFFER_PROCESS (2)
+			assert(gbBufferMsgs != BUFFER_PROCESS);
```
Comment (AJenbo) [inline Source/multi.cpp:469]:
```suggestion
			assert(gbBufferMsgs != 2);
```

File: Source/sync.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/sync.cpp]: That explains the odd comment in multi.cpp :)

**Review by glebm — APPROVED** [review]

**Pull request merged**